### PR TITLE
Change Credentials change

### PR DESCRIPTION
**Pull request by klsmith-usgs**

### CODE REVIEW

File: dbconnect.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import psycopg2
+
+
+class DBConnect(object):
+    """
+    Class for connecting to a postgresql database using a with statement
+    """
+
+    def __init__(self, dbhost='', db='', dbuser='', dbpass='', dbport=3306, autocommit=False,
```
Comment (dhill-usgs) [inline dbconnect.py]: 5432 now for pg instead of 3306?

File: dbconnect.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                 *args, **kwargs):
+        try:
+            self.conn = psycopg2.connect(host=dbhost, database=db, user=dbuser,
```
Comment (dhill-usgs) [inline dbconnect.py]: any way to rework this to use a context manager?  (the with statement?)  That way connections are never left hanging open.

File: dbconnect.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                                         password=dbpass, port=dbport)
+            self.cursor = self.conn.cursor()
+        except psycopg2.Error:
+            raise
+
+        self.autocommit = autocommit
+        self.fetcharr = []
+
+    def execute(self, sql_str):
```
Comment (dhill-usgs) [inline dbconnect.py]: accepting a straight SQL string for everything would open up the possibility of SQL injection attacks.  queries that accept parameters should always be parametarized through the cursor object... FYI.  Not saying that would become an issue here given where this class is located and how it's used, just making you aware.

http://stackoverflow.com/questions/1466741/parameterized-queries-with-psycopg2-python-db-api-and-postgresql
-

File: dbconnect.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        try:
+            self.cursor.execute(sql_str)
+        except psycopg2.Error:
+            raise
+
+        if self.autocommit:
+            self.commit()
+
+    def select(self, sql_str):
```
Comment (dhill-usgs) [inline dbconnect.py]: Same with sqlinjection.

File: dbconnect.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    def commit(self):
+        try:
+            self.conn.commit()
+        except psycopg2.Error:
+            raise
+
+    def rollback(self):
+        self.conn.rollback()
+
+    def __enter__(self):
```
Comment (dhill-usgs) [inline dbconnect.py]: ahhh, i see.  you set this up to be usable from a with statement, didn't ya?  ignore previous comment.

File: dbconnect.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.cursor.close()
```
Comment (dhill-usgs) [inline dbconnect.py]: do you need to check for null or None here first?

File: reporting/lsrd_collection_metrics.pig [hunk 7]
```diff
@@ -1,128 +0,0 @@
-/*
-  Name: lsrd_collection_metrics.pig
-
-  Description:  Pig script to run on the LSRD/ESPA hadoop cluster to
-                generate monthly metrics for GLS collection distribution
```
Comment (dhill-usgs) [inline reporting/lsrd_collection_metrics.pig]: Thank you.

File: reporting/lsrd_ondemand_metrics.pig [hunk 8]
```diff
@@ -1,84 +0,0 @@
-/*
```
Comment (dhill-usgs) [inline reporting/lsrd_ondemand_metrics.pig]: Thank you, thank you.

File: reporting/lsrd_stats.py [hunk 9]
```diff
@@ -1,675 +1,388 @@
...
+
+from dbconnect import DBConnect
+from utils import get_cfg, send_email
+
+
+FILE_PATH = os.path.realpath(__file__)
+LOG_FILE = '/data/logs/espa.cr.usgs.gov-access_log.1'
+
+# This info should come from a config file
+EMAIL_FROM = 'espa@espa.cr.usgs.gov'
```
Comment (dhill-usgs) [inline reporting/lsrd_stats.py]: configure this if time allows.

File: reporting/lsrd_stats.py [hunk 10]
```diff
@@ -1,675 +1,388 @@
...
+# EMAIL_TO = ['jenkerson@usgs.gov', 'lowen@usgs.gov', 'dhill@usgs.gov', 'klsmith@usgs.gov']
+EMAIL_TO = ['klsmith@usgs.gov']
```
Comment (dhill-usgs) [inline reporting/lsrd_stats.py]: same.

File: reporting/lsrd_stats.py [hunk 11]
```diff
@@ -1,675 +1,388 @@
...
+DEBUG_EMAIL_TO = ['klsmith@usgs.gov']
+EMAIL_SUBJECT = 'LSRD ESPA Metrics for {0} to {1}'
+
+ORDER_SOURCES = ('EE', 'ESPA')
+
+
+def arg_parser():
```
Comment (dhill-usgs) [inline reporting/lsrd_stats.py]: I found something really cool a while back that Morton is using, haven't used it myself yet.  Check this out: http://click.pocoo.org/5/arguments/

File: dbconnect.py (hunk 1 above)
Comment (klsmith-usgs) [inline dbconnect.py]: This get's overridden by what comes out of the .cfgnfo file. So, maybe it should be changed to the default PostgreSQL port, or changed to some None type.

File: reporting/lsrd_stats.py [hunk 12]
```diff
@@ -1,675 +1,388 @@
...
+
+    return args
+
+
+def setup_cron():
+    """
+    Setup cron job for the 1st of the month
+    for the previous month's metrics
+    """
+    chron_file = 'tmp'
```
Comment (dhill-usgs) [inline reporting/lsrd_stats.py]: typo?

File: reporting/lsrd_stats.py [hunk 13]
```diff
@@ -1,675 +1,388 @@
...
+
+    cron_str = '00 06 1 * * /usr/local/bin/python {1} -p'.format(FILE_PATH)
+
+    crons = subprocess.check_output(['crontab', '-l']).split('\n')
```
Comment (dhill-usgs) [inline reporting/lsrd_stats.py]: are there any libs out there for interacting with cron from python?  idk.

File: dbconnect.py (hunk 2 above)
Comment (klsmith-usgs) [inline dbconnect.py]: There is by wrapping it in another manager, but that was one of the purposes of this class. The __enter__ and __exit__ methods should provide the close-outs.

File: reporting/lsrd_stats.py [hunk 14]
```diff
@@ -1,675 +1,388 @@
...
+            return
+
+    add = ['#-----------------------------',
+           '# LSRD ESPA stats',
+           '#-----------------------------',
+           cron_str]
+
+    crons.extend(add)
+
+    with open(chron_file, 'w') as f:
```
Comment (dhill-usgs) [inline reporting/lsrd_stats.py]: please make one change... when writing a new cron, please backup current cron to the home directory of the user, name it backups/crontab-mmddyy-hhmmss

File: utils.py [hunk 15]
```diff
@@ -0,0 +1,49 @@
+import smtplib
+from email.mime.text import MIMEText
+
+
+CFG_NFO = "/home/{0}/.cfgnfo"
```
Comment (dhill-usgs) [inline utils.py]: one thing about the .cfgnfo... it actually is a standard formatted file that can be read with the Python ConfigParser lib.  I'm doing that in the espa-web project... (espa-web/web/ordering/espa_web/settings.py)

**Comment by dhill-usgs** [comment]
Only other comment is what we previously talked about... it's critical that the reported results are consistent with prior results.  Should be able to verify with the log file data now.

**Pull request closed without merging**